### PR TITLE
chore(uploader): remove unneeded `hasMultipleFiles` prop

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
This PR removes the external `hasMultipleFiles` API due to being redundant with the `maxFileCount` API. There isn't a scenario I could think of where these two props would need to be used together. `hasMultipleFiles={false}` is equivalent to `maxFileCount={1}` and `hasMultipleFiles={true}` is equivalent to `maxFileCount={2 to ♾)` or just leaving `maxFileCount` undefined.

This PR also renames `maxSize` to `maxFileSize` to be consistent with `maxFileCount` prop

#### Description of how you validated changes
* Smoke tested the example app and docs
* Ran unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/next/pages/ui/components/storage/file-uploader/index.page.tsx [hunk 1]
```diff
@@ -14,8 +14,7 @@ export default function FileUploaderEmail() {
       onSuccess={onSuccess}
       acceptedFileTypes={['image/*']}
       accessLevel="public"
-      hasMultipleFiles={true}
```
Comment (reesscot) [inline examples/next/pages/ui/components/storage/file-uploader/index.page.tsx:17]: This is actually a good example of why hasMultipleFiles is redundant. We shouldn't have to define both props here to get multi selection in the file uploader picker.

**Review by dbanksdesign — APPROVED** [review]

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/__tests__/useFileUploader.test.tsx [hunk 2]
```diff
@@ -85,11 +85,11 @@ describe('useFileUploader', () => {
     expect(result.current.fileStatuses.length).toEqual(1);
   });
 
-  it('returns 1 and updates file statuses when not hasMultipleFiles and target length is 1', () => {
+  it('returns 1 and updates file statuses when not allowMultipleFiles and target length is 1', () => {
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/__tests__/useFileUploader.test.tsx:88]: Off subject: Can we make a note to clean up these test descriptions

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/types.ts [hunk 3]
```diff
@@ -17,3 +17,14 @@ export interface UseFileUploader extends DragActionHandlers {
...
 }
+
+export interface UseFileUploaderProps {
+  maxFileSize: number;
+  acceptedFileTypes: string[];
+  allowMultipleFiles: boolean;
+  isLoading: boolean;
+}
+
+export interface UseFileUploader {
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/types.ts:28]: Nit: we use hook PascalCase for the return type for a fair amount of our existing hooks. Using `UseFileUploader` to type the function breaks convention

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.ts [hunk 4]
```diff
@@ -4,17 +4,12 @@ import { checkMaxSize, returnAcceptedFiles } from '@aws-amplify/ui';
 import { Files, FileState, FileStatuses } from '../../types';
 import { UseFileUploader } from './types';
 
-export default function useFileUploader({
-  maxSize,
+export const useFileUploader: UseFileUploader = ({
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.ts:7]: Prefer named functions for larger utils, and it better aligns with existing conventions as noted in the earlier comment about usage of PascalCase and hooks names

File: packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx [hunk 5]
```diff
@@ -63,6 +62,10 @@ export function FileUploader({
     ...overrideDisplayText,
   };
 
+  const allowMultipleFiles =
```
Comment (calebpollman) [inline packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx:65]: Wondering if we can remove this entirely and just infer what we need from `maxFileCount`? Might be easier to do later once the file state lifecycle has been refactored

**Review by calebpollman — APPROVED** [review]
Approved, but left some questions/feedback

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/__tests__/useFileUploader.test.tsx (hunk 2 above)
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/__tests__/useFileUploader.test.tsx:88]: Yes, will address in a subsequent pr

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/types.ts (hunk 3 above)
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/types.ts:28]: good catch, will update

File: packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx (hunk 5 above)
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx:65]: Yeah, I can revisit when I do the file state lifecycle.

File: packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.ts (hunk 4 above)
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.ts]: updated

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**